pkg/amber/provenance.go

```diff
@@ -89,6 +89,10 @@ func ParseProvenanceFile(path string) (*intoto.Statement, error) {
 		return nil, fmt.Errorf("could not unmarshal the provenance file:\n%v", err)
 	}
 
+	if len(statement.Subject) == 0 || statement.Subject[0].Digest["sha256"] == "" {
+		return nil, fmt.Errorf("at least one subject must be present, and it must have a sha256 digest")
```

Comment:
What would be the semantic of multiple subjects? Perhaps we should limit the check to ensure exactly one subject for now, until we figure that out (if it's even something we want to allow at all -- I suspect not).